### ISSUE TITLE
Add transformTZ

### ISSUE THREAD

**Issue by CapnRedB** [knowledgecode/date-and-time#56]
If you already have a date string, but in the wrong time zone, you have to transform, then parse and then format it back in order to get it to the correct timezone. It would be nice if there were a transformTZ option.

Though I understand the use case is rather limited and probably not necessary in most cases. My use case is the API I'm working with gives me a random timezone each time I call it. So I know the format before I even know the time. 🙃 

It would just save me an extra parse and format call. If there really isn't a need for it, don't worry about it.

**Comment by knowledgecode** [comment]
> If you already have a date string, but in the wrong time zone, you have to transform, then parse and then format it back in order to get it to the correct timezone.

Sorry, I can not image this situation. Could you show me a specific example with code?

**Comment by CapnRedB** [comment]
Sure! My code is a touch out of date. Had the "try" in there due to the +1300 bug that was just fixed

`		let initialTimeString = reformatChallongeTime(element.tournament['start_at']);
		initialTimeString['time'] = date.transform(initialTimeString['time'],challongeTimeFormat,desiredTimeFormat);
		let convertedTimezone = date.parseTZ(initialTimeString['time'], desiredTimeFormat,initialTimeString['UTC'], 'America/New_York');
		let finalTime;
		try {
		finalTime = date.formatTZ(convertedTimezone,desiredTimeFormat,'America/New_York');
		}
		catch (e){
			fs.writeFileSync('G:/Coding Projects/stdout/timeshit.txt','Initial Time String: ' +JSON.stringify(element.tournament['start_at'])+'\nPost Colon Removal: '+JSON.stringify(initialTimeString)+'\nConverted Time Zone: '+JSON.stringify(convertedTimezone)+'\n',{ flag: 'a' });
			throw e;
		}`

If transformTZ were a thing, I could just do that. But as it stands, transform will set the timezone back to local so I have to run it first (I was running it last).

**Comment by knowledgecode** [comment]
Um...I can't understand what you want to do because I'm not sure what are the specific value of these variables.

**Comment by knowledgecode** [comment]
Are the following interpretations correct?

You have a date time string with a timezone, and you want to convert this timezone to `America/New_York` (EST). For example, if you have a date time string like `2022-02-22T14:00:00.000+1300`, you want to finally get the string `2022-02-21T20:00:00.000-0500`. To do this, you currently need to write the following:

```JS
// This is a provided string.
const provided_string = '2022-02-22T14:00:00.000+1300';

// First, you need to convert the string to a Date object.
const provided_date = date.parse(provided_string, 'YYYY-MM-DD[T]HH:mm:ss.SSSZ');
// Then, you convert the Date object to a string using `formatTZ`.
const string_you_want = date.formatTZ(provided_date, 'YYYY-MM-DD[T]HH:mm:ss.SSSZ', 'America/New_York');

// string_you_want => 2022-02-21T20:00:00.000-0500
```

So you want a convenient function like `transformTZ` that makes it easy to write the above process.

```JS
const provided_string = '2022-02-22T14:00:00.000+1300';

const string_you_want = date.transformTZ(provided_string, 'YYYY-MM-DD[T]HH:mm:ss.SSSZ', 'America/New_York');

// string_you_want => 2022-02-21T20:00:00.000-0500
```

**Comment by CapnRedB** [comment]
Ah sorry for the delay. Let me send over the specifics. 
The formats I'm working with:
```js
const challongeTimeFormat = date.compile('YYYY-MM-DD[T]HH:mm:ss.SSSZ');
const desiredTimeFormat = date.compile('MMM D h:mmA [GMT]Z');
```
the API adds a colon in the middle of the timezone. Since date-and-time can only tokenize a 'Z' with no configuration, I have to remove that. This is what reformatChallongeTime does.
```js
//convert "2022-02-11T04:00:00.000+03:00" to "2022-02-11T04:00:00.000+0300"
let initialTimeString = reformatChallongeTime(element.tournament['start_at']);
```

It then goes:
```js
initialTimeString['time'] = date.transform(initialTimeString['time'],challongeTimeFormat,desiredTimeFormat);
let convertedTimezone = date.parseTZ(initialTimeString['time'], desiredTimeFormat,initialTimeString['UTC'], 'America/New_York');
let finalTime;
		try {
		finalTime = date.formatTZ(convertedTimezone,desiredTimeFormat,'America/New_York');
		}
```

What I noticed is that date.transform will take any input I give it and set it to my local timezone. So when I ran the date.transform last I got something like:

Feb 28 6:00PM GMT-0600

When running date.transform first I would get:

Feb 28 7:00PM GMT-0500

That's when I realized if there was just date.transformTZ I could do:

```js
initialTimeString['time'] = date.transform(initialTimeString['time'],challongeTimeFormat,desiredTimeFormat, 'America/New_York');
```
And there would be no need to parse and the format it back. Since I want to end with a date string and I start with a date string. The only way to go from String -> String with a specific timezone is to parseTZ (string -> dateObj) and then format (dateObj -> string). 

I hope that clears things up.

**Comment by knowledgecode** [comment]
Thank you, almost understand. I will consider implementing such a function.

**Comment by knowledgecode** [comment]
Added `tranformTZ()` to `timezone` plugin, also added `ZZ` token to support time zone offset values with colon (i.e. -08:00, +09:00). These changes have been included in the new version v2.2.0. Please let me know if you find any problems.
